configure.ac

```diff
@@ -61,8 +61,20 @@ case $host in
      lt_cv_deplibs_check_method="pass_all"
   ;;
 esac
-dnl Require C++11 compiler (no GNU extensions)
-AX_CXX_COMPILE_STDCXX([11], [noext], [mandatory], [nodefault])
+
+AC_ARG_ENABLE([c++17],
+  [AS_HELP_STRING([--enable-c++17],
+  [enable compilation in c++17 mode (disabled by default)])],
+  [use_cxx17=$enableval],
+  [use_cxx17=no])
+
+dnl Require C++11 or C++17 compiler (no GNU extensions)
+if test "x$use_cxx17" = xyes; then
+  AX_CXX_COMPILE_STDCXX([17], [noext], [mandatory], [nodefault])
```

Comment:
Given it's been removed in the `ax_cxx_compile_stdcxx.m4` macro update, the `[nodefault]` argument should be removed here and below.

Reply:
Done.

Reply:
The `[nodefault]` argument was added in _our_ patch: https://github.com/bitcoin/bitcoin/pull/9831/commits/9829c54de2725037ee0702306cbaa99fc9aa1826

Why is it possible to drop it?

Reply:
Ouch. I had no idea this was a local modification.

Reply:
Ah, it seems the new upstream version actually changed behaviour and now does what we called "nodefault" regardless.

Even with compilers that support c++17, if we request c++11, we'll only get c++11 (which can be verified from the Travis build logs).

Reply:
I should have made that more clear. I’ve got the same change as part of #18088.